toolkit/scripts/update_toolchain_manifest.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/python3
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+from os.path import dirname
+
+import argparse
+import shlex
+import subprocess
+
+# Can't rely on Python's 'pyrpm.spec' module - it's not as good with parsing the spec as 'rpmspec' and may leave unexpanded macros.
+RPMSPEC_COMMAND_COMMON = "rpmspec --parse -D 'forgemeta %{{nil}}' -D 'py3_dist X' -D 'with_check 0' -D 'dist .azl3' -D '__python3 python3' -D '_sourcedir {source_dir}' -D 'fillup_prereq fillup'"
+
+class Entry:
+    def __init__(self, name, version, release: str):
+        self.name = name
+        self.version = version
+        self.release = release
+
+    def get_processed_entry(self) -> str:
+        return self.version+"-"+self.release
+
+def formatted_rpmspec_command(spec_path: str) -> str:
+    source_dir = dirname(spec_path)
+    return f"{RPMSPEC_COMMAND_COMMON.format(source_dir=source_dir)}"
+
+def read_spec_name(spec_path: str) -> str:
+    return read_spec_tag(spec_path, "NAME")
+
+def read_spec_version(spec_path: str) -> str:
+    return read_spec_tag(spec_path, "VERSION")
+
+def read_spec_release(spec_path: str):
+    return read_spec_tag(spec_path, "RELEASE")
+
+def read_spec_tag(spec_path, tag: str) -> str:
+    command = formatted_rpmspec_command(spec_path)
+    raw_output = subprocess.check_output(shlex.split(f"{command} --srpm --qf '%{{{tag}}}' -q {spec_path}"),
```

Comment:
You should create the command args as an array of strings, instead of using `shlex.split`. This will avoid you needing to deal with string escaping.